File: Project Files/R/visualize.R
library(ggplot2)

plot_predictions <- function(svm_model, lr_model, X_test, y_test) {
  X <- X_test$days
  actual <- y_test

  df <- data.frame(
    days = X,
    actual = actual,
    svm_pred = predict(svm_model, data.frame(days = X)),
    lr_pred = predict(lr_model, data.frame(days = X))
  )

  ggplot(df, aes(x = days)) +
    geom_point(aes(y = actual), color = "black", size = 2, alpha = 0.7) +
    geom_line(aes(y = svm_pred), color = "red", linetype = "dashed", size = 1.2) +
    geom_line(aes(y = lr_pred), color = "blue", linetype = "solid", size = 1.2) +
    labs(
      title = "COVID-19 India: SVM vs Linear Regression",
      x = "Days",
      y = "Confirmed Cases"
    ) +
    theme_minimal() +
    theme(plot.title = element_text(hjust = 0.5)) +
    scale_y_continuous(labels = scales::comma)
}

File: Project Files/Python/visualize.py
import matplotlib.pyplot as plt

def plot_predictions(svm_model, lr_model, X_test, y_test):
    y_pred_svm = svm_model.predict(X_test)
    y_pred_lr = lr_model.predict(X_test)

    plt.figure(figsize=(10, 6))
    plt.scatter(X_test, y_test, color='black', label='Actual Data')
    plt.plot(X_test, y_pred_svm, color='red', label='SVM Predictions')
    plt.plot(X_test, y_pred_lr, color='blue', label='Linear Regression Predictions')
    plt.xlabel('Days')
    plt.ylabel('Confirmed Cases')
    plt.title('COVID-19 Cases Prediction for India 🇮🇳')
    plt.legend()
    plt.grid(True)
    plt.show()